modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -113,22 +104,31 @@ function AddFDSFilterModalContent({
 		getAllPicklists().then((items) => {
 			setPicklists(items);
 
-			const newVal = items.find(
+			const picklist = items.find(
 				(item) =>
 					String(item.externalReferenceCode) ===
 					(filter as any)?.listTypeDefinitionERC
 			);
 
-			if (newVal) {
-				setSelectedPicklist(newVal);
+			if (picklist) {
+				setSelectedPicklist(picklist);
 
-				setPreselectedValues(
-					newVal.listTypeEntries.filter((item) =>
+				const validSavedPreselectedValues = picklist.listTypeEntries.filter(
+					(item) =>
 						JSON.parse(
 							(filter as ISelectionFilter).preselectedValues ||
 								'[]'
 						).includes(item.externalReferenceCode)
-					)
+				);
+
+				setPreselectedValues(validSavedPreselectedValues);
+
+				setIncludeMode(
+					validSavedPreselectedValues?.length
+						? filter && (filter as ISelectionFilter).include
+							? 'include'
+							: 'exclude'
+						: 'include'
```

Comment:
I had to duplicate code, instead of keeping function, because this is also a `useEffect`. It's not a lot to duplicate, so I think it's a bit better than misleading `useCallback`.